// Copyright 2015, Google Inc.
// All rights reserved.


syntax = "proto3";

// interface de servi√ßo

service MusicServ {
  rpc getListMusics(ListRequest) returns (stream Music) {}
  rpc addMusic(Music) returns (BooleanReply) {}
  rpc deleteMusic(MusicId) returns (BooleanReply) {}
}

message MusicList {
  repeated Music music = 1;
}

message MusicId {
  int32 id = 1;
}

message Music {
    int32 id = 1;
    string title = 2;
    string band = 3;
    string genre = 4;

    repeated Music music = 5;
}

message BooleanReply {
    bool reply = 1;
}

message ListRequest {
    bool listMusics = 1;
}